src/components/sale/MemberPropertyModal.tsx

```diff
@@ -0,0 +1,241 @@
+  letter-spacing: 0.8px;
+  color: var(--gray-darker);
+`
+const StyledContent = styled.div`
+  font-size: 16px;
+  font-weight: 500;
+  line-height: 2;
+  letter-spacing: 0.2px;
+  color: var(--gray-darker);
+`
+const StyledLink = styled.a`
+  margin-left: 5px;
+  color: ${props => props.theme['@primary-color'] || '#4c5b8f'};
+`
+
+type FieldProps = {
+  property: { [propertyId: string]: string }
+  note: {
+    status?: string
+    description?: string
+  }
+}
+
+const MemberPropertyModal: React.FC<
+  AdminModalProps & {
+    member: {
+      id: string
+      name: string
+      categoryNames: string[]
+    } | null
+    onSubmit?: (values: FieldProps) => void
+    onRefetch?: () => void
+  }
+> = ({ member, onSubmit, onRefetch, ...modalProps }) => {
+  const { formatMessage } = useIntl()
+  const [form] = useForm<FieldProps>()
+  const { loadingProperties, properties } = useProperty()
+  const { loadingMemberProperties, memberProperties } = useMemberPropertyCollection(member?.id || '')
+  const { updateMemberProperty } = useMutateMemberProperty()
+  const { loading: loadingMemberNoteStatus, data: memberNoteStatus } = useQuery<
+    hasura.GET_LATEST_MEMBER_NOTE_STATUS,
+    hasura.GET_LATEST_MEMBER_NOTE_STATUSVariables
+  >(GET_LATEST_MEMBER_NOTE_STATUS, { variables: { memberId: member?.id || '' } })
+  const [loading, setLoading] = useState(false)
+
+  const displayProperties = [
+    { name: '性別', isRequired: true },
+    { name: '縣市', isRequired: false },
+    { name: '有意願領域', isRequired: true },
+    { name: '是否在職', isRequired: true },
+    { name: '是否為相關職務', isRequired: false },
+    { name: '學生程度', isRequired: true },
+    { name: '學習動機', isRequired: false },
+    { name: '每月學習預算', isRequired: false },
+    { name: '有沒有上過其他課程', isRequired: false },
+    { name: '是否有轉職意願', isRequired: true },
+  ]
+
+  const handleSubmit = () => {
+    if (!member) {
+      return
+    }
+    form
+      .validateFields()
+      .then(values => {
+        updateMemberProperty({
+          variables: {
+            memberId: member.id,
+            memberProperties: properties.map(property => {
+              const displayPropertiesName = displayProperties.map(displayProperty => displayProperty.name)
+              if (displayPropertiesName.includes(property.name)) {
+                return { member_id: member.id, property_id: property.id, value: values.property[property.id] || '' }
+              }
+
+              return {
+                member_id: member.id,
+                property_id: property.id,
+                value: memberProperties.find(memberProperty => memberProperty.id === property.id)?.value || '',
+              }
+            }),
+          },
+        })
+          .then(() => {
+            message.success(formatMessage(commonMessages.event.successfullySaved))
+            onSubmit?.(values)
+            onRefetch?.()
+          })
+          .catch(handleError)
+          .finally(() => setLoading(false))
+      })
```

Comment:
`updateMemberProperty` can move to props onSubmit to avoid nesting promise

```ts
onSubmit={values => {
  return updateMemberProperty({})
    .then(() => {
      if (values.note?.status && values.note?.description) {
        insertMemberNote({
          variables: {
            memberId: selectedMember.id,
            authorId: sales.id,
            status: values.note.status,
            description: values.note.description,
            duration: 0,
          },
        })
      }
    })
    .then(() => {
      onRefetch?.()
    })
    .catch(handleError)
}}
```

Reply:
great reminding , fixed！